python/orca/tutorial/pytorch/transfer_learning/train.py

```diff
@@ -0,0 +1,278 @@
+# Most of the pytorch code is adapted from PyTorch's transfer learning tutorial for
+# hymenoptera dataset.
+# https://pytorch.org/tutorials/beginner/transfer_learning_tutorial.html
+#
+
+# Step 0: Import necessary libraries
+import os
+import numpy as np
+import matplotlib.pyplot as plt
+
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from torch.optim import lr_scheduler
+import torchvision
+from torchvision import datasets, transforms, models
+
+from bigdl.orca import init_orca_context, stop_orca_context
+from bigdl.orca.learn.pytorch import Estimator
+from bigdl.orca.learn.metrics import Accuracy
+
+plt.ion()  # interactive mode
+
+
+# Step 1: Init Orca Context
+sc = init_orca_context()
+
+
+# Step 2: Define train and test datasets as PyTorch DataLoader
+def load_dataset(dataset_dir, batch_size=4):
+    # Data augmentation and normalization for training
+    # Just normalization for validation
+    data_transforms = {
+        'train': transforms.Compose([
+            transforms.RandomResizedCrop(224),
+            transforms.RandomHorizontalFlip(),
+            transforms.ToTensor(),
+            transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])
+        ]),
+        'val': transforms.Compose([
+            transforms.Resize(256),
+            transforms.CenterCrop(224),
+            transforms.ToTensor(),
+            transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])
+        ]),
+    }
+
+    image_datasets = {x: datasets.ImageFolder(os.path.join(data_dir, x),
+                                              data_transforms[x])
+                      for x in ['train', 'val']}
+    dataset_sizes = {x: len(image_datasets[x]) for x in ['train', 'val']}
+    class_names = image_datasets['train'].classes
+
+    dataloaders = {x: torch.utils.data.DataLoader(image_datasets[x], batch_size=batch_size,
+                                                  shuffle=True, num_workers=1)
+                   for x in ['train', 'val']}
+    train_loader = dataloaders['train']
+    test_loader = dataloaders['val']
+    return train_loader, test_loader, class_names
+
+
+def imshow(inp, title=None):
+    """
+    Visualize a few training images
+    so as to understand the data augmentations.
+    """
+    inp = inp.numpy().transpose((1, 2, 0))
+    mean = np.array([0.485, 0.456, 0.406])
+    std = np.array([0.229, 0.224, 0.225])
+    inp = std * inp + mean
+    inp = np.clip(inp, 0, 1)
+    plt.imshow(inp)
+    if title is not None:
+        plt.title(title)
+    plt.pause(0.001)  # pause a bit so that plots are updated
+
+
+def train_loader_func(config, batch_size):
+    train_loader, test_loader, class_names = load_dataset(config['data_dir'], batch_size)
+
+    inputs, classes = next(iter(train_loader))  # Get a batch of training data
+    out = torchvision.utils.make_grid(inputs)  # Make a grid from batch
+    imshow(out, title=[class_names[x] for x in classes])

```

Comment:
Run this code segment outside of the func on the driver. This function will be executed on executors and you won't be able to see the figures.